tools/ncf_constraints.py

```diff
@@ -0,0 +1,111 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+if sys.version_info < (3,):
+  string_type = unicode
+else:
+  string_type = str
+
+### Constraint checking function
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
```

Comment:
I don't think we should use re.M

I think we want to check the whole value with our regex, so with you first case you would match 
    A
    A
    B

If you use re.M and would not match with other cases, and i think we don't want to match